### ISSUE TITLE
igx-data-chart memory leak when there are two or more charts with the same sync channel

### ISSUE THREAD

**Issue by jsakamotoIGJP** [IgniteUI/igniteui-angular#13995]
When two or more igx-data-chart components on a page are combined with the same sync channel, a memory leak occurs after leaving that page.

## Steps to reproduce

1. Open the browser's Dev Tools / Memory tab.
2. Take the 1st memory snapshot (don't forget to click the "Collect garbage" button before taking.)
3. Just navigate to the page with two igx-data-chart components combined with the same sync channel.
4. Go back to the home page.
5. Take the 2nd memory snapshot (remember to click the "Collect garbage" button before taking it).
6. In the Dev Tools Memory tab, Choose "Comparison" in the "Perspective" dropdown and input "igx" into the "Class filter" text field.

https://github.com/IgniteUI/igniteui-angular/assets/102948935/aa518e69-6953-4ac9-8b0f-6c78a1d980e8

## Result  
You will see there are some objects with the name "igx" that were added but not deleted in the "Comparison" view of the Memory tab.

## Expected result  
The "Delta" column for all objects with the name "igx" should be zero in the "Comparison" view of the Memory tab.

## Attachments  

[📦c-00235192-ng-app1-v17.1.4 (3).zip](https://github.com/IgniteUI/igniteui-angular/files/14632940/c-00235192-ng-app1-v17.1.4.3.zip)

**Comment by gmurray81** [comment]
Please note we don’t technically consider this to be a bug. It is intended that the syncchannel keeps a chart pinned and prevents it from being collected. There are expensive resources associated with the synclink if you are simply detaching the chart from the layout and moving it elsewhere that we don’t wish to churn.
However, for Angular, since we do receive an unambiguous signal that a component is being destroyed, we will unhook the chart from the syncchannel when angular sends this signal, as a resolution.
But it is intended generally that a user remove a chart from a synchannel when it is being discarded. On other platforms we do not receive unambiguous signals that you are finished with the component.

**Comment by jsakamotoIGJP** [comment]
@gmurray81 I strongly agree that it is not technically a bug. On the other hand, is there a workaround for this issue, such as an explicit way to release resources? Or is it okay to request a workaround, such as creating a workaround entry point? Or, according to your reply, can I expect that this issue is technically solvable in Angular at least, and that the fixed version of "Ignite UI for Angular" will be released someday?

Because regardless of how this issue is from a technical standpoint, we need to be able to provide some solutions or workarounds to a support request from a customer who is having trouble with memory leaks.

Your feedback is greatly appreciated as we work together to find a solution to this issue.

**Comment by gmurray81** [comment]
@jsakamotoIGJP sorry, I was trying to explain that above. It should be fine as long as you remove it from the `syncChannel`, set that to null and it should unpin the resource. Anyhow, as I said we might make angular in particular clean up since it gets an explicit destory message. But we weren't prioritizing it given you can explicitly release by nulling the syncChannel.

**Comment by jsakamotoIGJP** [comment]
@gmurray81 Thank you for your so prompt response!
Oh, sorry for my lack of reading skills. Okay, I believe I understand now. I'll respond to our customers to explain this method of explicitly removing charts from syncChannel. I'll let you know the result of our communication with our customers later.
Thanks!
